Scripts/AlphabetMapper.cs

```diff
@@ -1,50 +1,94 @@
 ﻿using Godot;
+using System;
 using System.Collections.Generic;
 
-/*
- * ToDo:
- * This wont be static in the future and will be instanciated at starting a New Game.
- * 
- * ToDo:
- * Modify the atlas to include fixed special characters?
- * 
- * ToDo:
- * Add learnt character to the algorithm.
- */
 public static class AlphabetMapper
 {
-    private const string _lettersAndSpecialChars = "abcdefghijklmnopqrstuvwxyz.:;,?!-()";
-    private const int _xMaxAtlas = 20;
-    private const int _yMaxAtlas = 20;
+    public const string LettersAndSpecialChars = "abcdefghijklmnopqrstuvwxyz.:;,?!-()";
 
-    private static Dictionary<char, Vector2I> _mapping = new Dictionary<char, Vector2I>(_lettersAndSpecialChars.Length);
+    public const int AlienAtlasIndex = 0;
+    public const int AlienAtlasDimension = 20;
+    
+    public const int OriginalAtlasIndex = 99;
+    public const int OriginalAtlasDimension = 6;
+
+    private static readonly Dictionary<char, AtlasVector> _alienMapping = new(LettersAndSpecialChars.Length);
+    private static readonly Dictionary<char, AtlasVector> _learningProgress = new(LettersAndSpecialChars.Length);
 
     public static void GenerateMapping(ulong seed)
     {
-        _mapping.Clear();
+        _alienMapping.Clear();
+        _learningProgress.Clear();
 
-        var r = new RandomNumberGenerator();
-        r.Seed = seed;
+        var rand = new RandomNumberGenerator
+        {
+            Seed = seed
+        };
 
-        var duplicationChecker = new HashSet<Vector2I>(_lettersAndSpecialChars.Length);
+        var duplicationChecker = new HashSet<Vector2I>(LettersAndSpecialChars.Length);
 
-        foreach (var i in _lettersAndSpecialChars)
+        foreach (var i in LettersAndSpecialChars)
         {
-            Vector2I vector;
-            do
-            {
-                vector = new Vector2I(r.RandiRange(0, _xMaxAtlas - 1), r.RandiRange(0, _yMaxAtlas - 1));
-            }
-            while (duplicationChecker.Contains(vector));
+            Vector2I vector = GetRandomUnique((v) => duplicationChecker.Contains(v), () => new Vector2I(rand.RandiRange(0, AlienAtlasDimension - 1), rand.RandiRange(0, AlienAtlasDimension - 1)));
 
             duplicationChecker.Add(vector);
-            _mapping.Add(i, vector);
+            _alienMapping.Add(i, new AtlasVector(AlienAtlasIndex, vector));
+        }
+    }
+
+    public static void LearnRandomCharacter()
+    {
+        // BUG: If everything has been learnt, infinite loop occours
```

Comment:
Add this:
```
if (_learningProcess.Count == LettersAndSpecialCharacters.Length)
  return;
```

because then learning was finished and no more random learning can occur.